SnapkitPractice2/SnapkitPractice2/ViewController.swift

```diff
@@ -0,0 +1,160 @@
+        rightTopStackView.addArrangedSubview(rightTopButton3)
+        
+        bottomStackView.addArrangedSubview(bottomLabel)
+        bottomStackView.addArrangedSubview(bottomLabel2)
+        bottomStackView.addArrangedSubview(bottomLabel3)
+        bottomStackView.addArrangedSubview(bottomLabel4)
+        bottomStackView.addArrangedSubview(bottomLabel5)
+        
+        self.view.addSubview(leftTopStackView)
+        self.view.addSubview(rightTopStackView)
+        self.view.addSubview(centerView)
+        self.view.addSubview(centerView2)
+        self.view.addSubview(bottomStackView)
+        
+        //왼쪽 상단 스택 뷰
+        leftTopStackView.axis = .vertical
+        leftTopStackView.spacing = 20
+        leftTopStackView.backgroundColor = .gray
+        leftTopStackView.snp.makeConstraints {
+            $0.leading.equalTo(view.safeAreaLayoutGuide.snp.leading).offset(30)
+            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(30)
+            $0.trailing.equalTo(rightTopStackView.snp.leading).offset(-30)
+            $0.bottom.equalTo(centerView.snp.top).offset(-30)
+        }
+        
+        leftTopView.backgroundColor = .red
+        leftTopView.snp.makeConstraints {
+            $0.height.equalTo(leftTopView.snp.width)
+        }
+        
+        leftTopView2.backgroundColor = .orange
+        leftTopView2.snp.makeConstraints {
+            $0.height.equalTo(leftTopView2.snp.width).dividedBy(2)
+        }
+        
+        leftTopView3.backgroundColor = .yellow
+        leftTopView3.snp.makeConstraints {
+            $0.height.equalTo(leftTopView3.snp.width).dividedBy(3)
+        }
+        leftTopView4.backgroundColor = .green
+        leftTopView4.snp.makeConstraints {
+            $0.height.equalTo(leftTopView4.snp.width).dividedBy(4)
+        }
+        
+        //오른쪽 상단 스택 뷰
+        rightTopStackView.axis = .horizontal
+        rightTopStackView.spacing = 20
+        rightTopStackView.backgroundColor = .gray
+        rightTopStackView.distribution = .fillEqually
+        
+        rightTopStackView.snp.makeConstraints{
+            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(30)
+            $0.height.equalTo(leftTopView.snp.height).dividedBy(2)
+            $0.leading.equalTo(leftTopStackView.snp.trailing).offset(30)
+            $0.trailing.equalTo(view.safeAreaLayoutGuide.snp.trailing).offset(-30)
+        }
+        
+        rightTopButton.backgroundColor = .purple
+        rightTopButton.setTitle("버튼1", for: .normal)
+        rightTopButton2.backgroundColor = .purple
+        rightTopButton2.setTitle("버튼2", for: .normal)
+        rightTopButton3.backgroundColor = .purple
+        rightTopButton3.setTitle("버튼3", for: .normal)
+        
+        //센터 뷰
+        centerView.backgroundColor = .systemBlue
+        
+        let centerViewRatio:Double = 8/15
+        centerView.snp.makeConstraints {
+            $0.leading.equalTo(view.safeAreaLayoutGuide.snp.leading).offset(30)
```

Comment:
leading, trailing의 경우 safeArea와 View의 위치 차이가 없으므로 아래와 같이 작성하시면 됩니다.

```suggestion
            $0.leading.equalTo(view.snp.leading).offset(30)
```